lib/bundle/brew_installer.rb

```diff
@@ -31,6 +32,10 @@ def install_change_state!
       end
     end
 
+    def start_service?
+      @start_service
+    end
+
     def restart_service?
       # Restart if `restart_service: :always`, or if the formula was installed or upgraded
       @restart_service && (@restart_service.to_s != 'changed' || changed?)
```

Comment:
Ok, let see if i understand you correctly.

The `restart_service?` now use both for installing and checking. But now the logic of them is different (with this PR). So we return the `restart_service?` to it old behaviour, which is return `@restart_service` directly, and add a new `restart_service_needed?` to determine if we need to restart the service or not when install or upgrade it.